system/core/events.php

```diff
@@ -67,15 +61,18 @@ function addEventHandler($html) {
 		url - the hypha-formatted page identifier, e.g. 'en/hompage/defaultview'. When left empty the current paged is reloaded.
 		cmd - php function to execute, e.g. 'save'
 		arg - argument to pass to the cmd function
+		callback - optional callback function for making ajax calls
 	*/
-	function hypha(url, cmd, arg) {
+	function hypha(url, cmd, arg, callback = null) {
```

Comment:
Adding support for a callback here should probably be a separate commit (not a separate PR, though).

I also wonder if adding a callback here is clear enough, since now adding the callback argument in a call completely changes the behaviour of this function. I wonder if it would be better to add the cmd and arg arguments to the `ajax` function (or add a flavour of that function with those arguments)?

Reply:
Not sure. In both cases an action is triggered on the client (probably by a user clicking something), and the some data gets routed to a specific piece of code on the server. The only difference is whether the outcome is a fresh page render (by means of a reload) or dynamically handling the outcome in the page that already exists on the client (through ajax). So although the behaviour is somewhat different the purpose (following up on client events) is pretty much the same. From that perspective introducing a separate command adds just as much confusion as it resolves. Maybe up for discussion another time, for now I would argue to adopt this approach. Will make it a separate commit though.

Reply:
Ok, let's leave it like this for now. We might need to revisit this as part of the newtext refactor anyway, since form handling and command routing is changed there anyway.